### PR TITLE
fixing json <=> json

### PR DESCRIPTION
**Pull request by paxtonhare**

### CODE REVIEW

**Comment by popzip** [comment]
@dmcassel will this get reviewed/merged during this sprint or after @paxtonhare back?

File: src/main/ml-modules/ext/com.marklogic.smart-mastering/survivorship/merging/base.xqy [hunk 1]
```diff
@@ -608,14 +610,37 @@ declare function merge-impl:properties-are-equal(
...
-  (:
-   : $map1 - $map2: find key-value pairs that exist in $map1 but not $map2
-   : $map1 + $map2: find the union of key-value pairs in the two maps
-  :)
-  map:count($object1 - $object2) = 0 and
-    map:count($object1 + $object2) = map:count($object1)
+  merge-impl:objects-equal-recursive($object1, $object2)
+};
+
+declare function merge-impl:objects-equal-recursive($object1, $object2) as xs:boolean
```
Comment (dmcassel) [inline src/main/ml-modules/ext/com.marklogic.smart-mastering/survivorship/merging/base.xqy]: declaring a return type disables tail recursion: https://docs.marklogic.com/guide/xquery/programming#id_93922

File: src/test/ml-modules/root/test/suites/merging-json/rollback-merge.xqy [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  let $s2-dt := $merged-doc//*:sources[*:name = "SOURCE2"]/*:dateTime/fn:string()
+  let $expected := xdmp:to-json(xdmp:from-json-string('{"envelope":{"headers":{"id":"' || $smid || '","merges":[{"document-uri":"/source/1/doc1.json"},{"document-uri":"/source/2/doc2.json"}],"sources":[{"name":"SOURCE2","import-id":"mdm-import-b96735af-f7c3-4f95-9ea1-f884bc395e0f","user":"admin","dateTime":"' || $s2-dt || '"},{"name":"SOURCE1","import-id":"mdm-import-8cf89514-fb1d-45f1-b95f-8b69f3126f04","user":"admin","dateTime":"' || $s1-dt || '"}]}, "instance":{"MDM":{"Person":{"PersonType":{"Address":{"AddressType":{"AddressPrivateMailboxText":"45","AddressSecondaryUnitText":"JANA","LocationCityName":"SCRANTON","LocationPostalCode":"18505","LocationState":"PA"}}, "CaseAmount": 1287.9, "CaseStartDate":"20110406","CustomThing":["2","1"],"PersonBirthDate":"19801001","PersonName":{"PersonNameType":{"PersonGivenName":"LINDSEY","PersonSurName":"JONES"}}, "PersonSSNIdentification":{"PersonSSNIdentificationType":{"IdentificationID":"393225353"}}, "PersonSex":"F","Revenues":{"RevenuesType":{"Revenue":"4332"}}, "id":["6270654339","6986792174"]}}}}}}'))
```
Comment (dmcassel) [inline src/test/ml-modules/root/test/suites/merging-json/rollback-merge.xqy]: kinda gross having this JSON here in a string. any reason not to put it in a file under test-data?

File: src/test/ml-modules/root/test/suites/merging-xml/test-data/custom-sjs-merge-options.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<options xmlns="http://marklogic.com/smart-mastering/merging">
+  <match-options>basic</match-options>
+  <thresholds>
```
Comment (dmcassel) [inline src/test/ml-modules/root/test/suites/merging-xml/test-data/custom-sjs-merge-options.xml]: merge options should no longer have thresholds: https://project.marklogic.com/jira/browse/MDM-298

File: src/test/ml-modules/root/test/suites/merging-xml/test-data/custom-xqy-merge-options.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<options xmlns="http://marklogic.com/smart-mastering/merging">
+  <match-options>basic</match-options>
+  <thresholds>
```
Comment (dmcassel) [inline src/test/ml-modules/root/test/suites/merging-xml/test-data/custom-xqy-merge-options.xml]: merge options should not have thresholds

File: src/test/ml-modules/root/test/suites/merging-json/rollback-merge.xqy (hunk 2 above)
Comment (paxtonhare) [reply src/test/ml-modules/root/test/suites/merging-json/rollback-merge.xqy]: because I have to dynamically insert values into it

**Pull request merged**